code/scripting/api/objs/ship.cpp

```diff
@@ -2337,6 +2338,147 @@ ADE_FUNC(setGlowPointBankActive, l_Ship, "boolean active, [number bank]", "Activ
+	}
+	else
+	{
+		docker_point = 0;
+	}
+
+	if (lua_isnumber(L, 4))
+	{
+		ade_get_args(L, "***i", &dockee_point);
+		dockee_point--;	// Lua --> C/C++
+	}
+	else if (lua_isstring(L, 4))
+	{
+		const char* name = nullptr;
+		ade_get_args(L, "***s", &name);
+		dockee_point = model_find_dock_name_index(Ship_info[dockee_shipp->ship_info_index].model_num, name);
+	}
+	else
+	{
+		dockee_point = 0;
+	}
+
+	if (docker_point < 0 || docker_point >= docker_pm->n_docks)
+	{
+		Warning(LOCATION, "Invalid docker point specified for ship '%s'", docker_shipp->ship_name);
+		return ADE_RETURN_NIL;
+	}
+
+	if (dockee_point < 0 || dockee_point >= dockee_pm->n_docks)
+	{
+		Warning(LOCATION, "Invalid dockee point specified for ship '%s'", dockee_shipp->ship_name);
+		return ADE_RETURN_NIL;
+	}
+
+	//Make sure that the specified dockpoints are all free (if not, do nothing)
+	if (dock_find_object_at_dockpoint(docker_objh->objp, docker_point) != nullptr ||
+		dock_find_object_at_dockpoint(dockee_objh->objp, dockee_point) != nullptr)
+	{
+		nprintf(("scripting", "Cannot dock %s to %s because at least one of the specified dockpoints is occupied", docker_shipp->ship_name, dockee_shipp->ship_name));
+		return ADE_RETURN_NIL;
+	}
+
+	//Set docked
+	dock_orient_and_approach(docker_objh->objp, docker_point, dockee_objh->objp, dockee_point, DOA_DOCK_STAY);
+	ai_do_objects_docked_stuff(docker_objh->objp, docker_point, dockee_objh->objp, dockee_point, true);
+
+	return ADE_RETURN_NIL;
+}
+
+static int jettison_helper(lua_State *L, object_h *docker_objh, float jettison_speed, int skip_args)
+{
+	object_h *dockee_objh = nullptr;
+	int num_ship_args = 0;
+
+	while (true)
+	{
+		// read the next ship
+		internal::Ade_get_args_skip = skip_args++;
+		if (ade_get_args(L, "|o", l_Ship.GetPtr(&dockee_objh)) == 0)
+			break;
+		num_ship_args++;
+
+		// make sure ship exists
+		if (dockee_objh == nullptr || !dockee_objh->IsValid())
+			continue;
+
+		// make sure we are docked to it
+		if (!dock_check_find_direct_docked_object(docker_objh->objp, dockee_objh->objp))
+			continue;
+
+		object_jettison_cargo(docker_objh->objp, dockee_objh->objp, jettison_speed, true);
+	}
+
+	// if we didn't find any specified ships, then we need to jettison all of them
+	if (num_ship_args == 0)
+	{
+		// Goober5000 - as with ai_deathroll_start, we can't simply iterate through the dock list while we're
+		// undocking things.  So just repeatedly jettison the first object.
+		while (object_is_docked(docker_objh->objp))
+		{
+			object_jettison_cargo(docker_objh->objp, dock_get_first_docked_object(docker_objh->objp), jettison_speed, true);
+		}
+	}
+
+	return ADE_RETURN_NIL;
+}
+
+ADE_FUNC(setUndocked, l_Ship, "[ship dockee_ship]", "Immediately undocks this ship from another ship or from an arbitrary number of ships - this function can accept an arbitrary number of ship arguments.", nullptr, "Returns nothing")
```

Comment:
This docstring is ambiguous in regards to "this / another ship", especially considering that usually, "this" would need to refer to the object on which the function is called . I suggest both this and jettison to be reworded like "undocks / jettisons dockee from this ship". Also, the docstring needs to explictly mention the special mode for no ships listed, as well as specify varargs, like ``[ship... dockee_ships /* All docked ships by default */]``